cmd/agent/subcommands/status/command.go

```diff
@@ -126,8 +104,17 @@ func redactError(unscrubbedError error) error {
 	return scrubbedError
 }
 
-func statusCmd(_ log.Component, config config.Component, _ sysprobeconfig.Component, cliParams *cliParams) error {
-	return redactError(requestStatus(config, cliParams))
+func statusCmd(logger log.Component, config config.Component, _ sysprobeconfig.Component, cliParams *cliParams) error {
+	if len(cliParams.args) < 1 {
+		return redactError(requestStatus(config, cliParams))
+	}
+
+	// TODO: remove in 7.54 release
+	if cliParams.args[0] == "component" {
+		fmt.Fprintf(os.Stderr, "[DEPRECATION WARNING] `datadog-agent status component [name]` syntax will be replace by `datadog-agent status [name]` in Datadog Agent 7.55\n")
+		cliParams.args = cliParams.args[1:]
```

Comment:
Seems a biut strange to manipulate the cliParams directly but if there's no side-effects, it may be ok

Reply:
It seems `cliParams` is only a string slice so It should be transparent for underlying functions.